judgels-frontends/raphael/src/components/SubmissionDetails/SubmissionDetails.tsx

```diff
@@ -257,7 +259,9 @@ export class SubmissionDetails extends React.PureComponent<SubmissionDetailsProp
         <h5>
           {key === 'source' ? '' : key + ': '} {source.submissionFiles[key].name}
         </h5>
-        <pre>{base64.decode(source.submissionFiles[key].content)}</pre>
+        <SyntaxHighlighter language="cpp" style={prism}>
```

Comment:
Hardcoded `cpp`? You can get the grading language via `submission.gradingLanguage` (you have to do mapping from the language to a valid syntax highlighter value). See `src/modules/api/gabriel/language.ts` for the list of possible languages.